src/Week2/LogAnalysis.hs

```diff
@@ -0,0 +1,63 @@
+{-# OPTIONS_GHC -Wall #-}
+module LogAnalysis where
+
+import Text.Read
+import Week2.Log
+
+---------- ex 1 ----------
+
+parseError :: String -> LogMessage
+parseError string =
+        readError $ words string
+    where
+
+        helper :: MessageType -> [String] -> LogMessage
+        helper err (t:rest) =
+            case (readMaybe t :: Maybe Int) of
+                Nothing  -> Unknown string
+                Just int -> LogMessage err int $ unwords rest
+        helper _ _ = Unknown string
+
+        readError :: [String] -> LogMessage
+        readError ("I":xs) = helper Info xs
+        readError ("W":xs) = helper Warning xs
+        readError ("E":n:xs) =
```

Comment:
You can avoid writing some helper functions if you do pattern matching just in the list. So something like:

```haskell
case words string of
    "I":n:xs ->
    "W":n:xs ->
    "E":n:m:xs ->
```